crates/consensus/src/transaction/deposit.rs

```diff
@@ -123,10 +123,10 @@ impl TxDeposit {
         self.source_hash.encode(out);
         self.from.encode(out);
         self.to.encode(out);
-        if let Some(mint) = self.mint {
-            mint.encode(out);
-        } else {
+        if self.mint == 0 {
             out.put_u8(EMPTY_STRING_CODE);
+        } else {
+            self.mint.encode(out);
         }
```

Comment:
I think we no longer have to do this because rlp(u128::Zero) is the same as EMPTY_STRING_CODE